### PR TITLE
Issue/unicode error

### PR DESCRIPTION
**Pull request by christinestraub**
This PR adds functionality to try other common encodings if an error related to the encoding is raised and the user has not specified an encoding.

Files changed:
> unstructured>file_utils>encoding.py (created)
> unstructured>partition>text.py (updated)
> test_unstructured>partition>test_text.py (updated)

### Testing

```
from unstructured.partition.text import partition_text

# try with filename
elements = partition_text(filename="example-docs/fake-text-utf-16.txt")
print("\n\n".join([str(el) for el in elements]))

# try with file-like object
with open("example-docs/fake-text-utf-16.txt") as f:
    elements = partition_text(file=f)
    print("\n\n".join([str(el) for el in elements]))

with open("example-docs/fake-text-utf-16.txt", "rb") as f:
    elements = partition_text(file=f)
    print("\n\n".join([str(el) for el in elements]))
```

```
@pytest.mark.parametrize(
    "filename",
    ["fake-text-utf-16.txt", "fake-text-utf-16-le.txt", "fake-text-utf-32.txt"],
)
def test_partition_text_from_filename_without_encoding(filename):
    filename = os.path.join(DIRECTORY, "..", "..", "example-docs", filename)
    elements = partition_text(filename=filename)
    assert len(elements) > 0
    assert elements == EXPECTED_OUTPUT
```

### CODE REVIEW

**Review by cragwolfe — APPROVED** [review]
It's a bit awkward that [auto.py](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/auto.py) still has the default encoding of `utf-8`, though that applies to other document types besides text.

What are your thoughts about the other file types? (html, pdf, etc)

For `partition_text`, LGTM!

**Review by cragwolfe — CHANGES_REQUESTED** [review]
Missed a changelog issue.

If cutting a new release, the next version is actually 0.6.9.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## 0.6.10
+
+### Enhancements
+
+### Features
+
+### Fixes
+
+* Adds functionality to try other common encodings if an error related to the encoding is raised and the user has not specified an encoding.
```
Comment (cragwolfe) [inline CHANGELOG.md:9]: so, two options on this changelog. cut a new pypi release at `0.6.9`, or just increment `0.6.9-dev1` to `0.6.9-dev2` and add the bullet point inline. in either case, the block under what is currently line 11 is the block to add the bullet item.

Comment (christinestraub) [reply CHANGELOG.md:9]: > It's a bit awkward that [auto.py](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/auto.py) still has the default encoding of utf-8, though that applies to other document types besides text.

> What are your thoughts about the other file types? (html, pdf, etc)


For now, the auto.py applies the default encoding of utf-8 to 5 types of files (txt, pdf, eml, html, xml)

`txt`: already implemented

`pdf`: The encoding kwarg is passed to partition_pdf, but is not used. (https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/pdf.py#L267) so I think the encoding kwarg should be removed from partition_pdf. 

`eml`: The .eml files have encoding information in the metadata. The encoding information is typically specified in the Content-Type header of the email message. (https://github.com/Unstructured-IO/unstructured/blob/main/example-docs/email-with-image.eml#L13). 

I think the encoding information should be parsed from the metadata if the user has not specified an encoding.

`html`: two options for filename and file-like object, except url
	option1: parse encoding information from html file metadata (<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">)
	option2: implement same pattern as partition_text

	factor to consider: whether to accept html files don't have charset information in the metadata

`xml`: similar to hmtl
	option1: parse encoding information from html file metadata (<?xml version="1.0" encoding="UTF-8"?>)
	option2: implement same pattern as partition_text

	factor to consider: whether to accept xml files don't have encoding information in the metadata

Comment (christinestraub) [reply CHANGELOG.md:9]: I'm not sure if I should do all the work for the 5 types of files in the same 'PR' or if I should create separate 'PRs'.

**Comment by christinestraub** [comment]
> It's a bit awkward that [auto.py](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/auto.py) still has the default encoding of utf-8, though that applies to other document types besides text.

> What are your thoughts about the other file types? (html, pdf, etc)


For now, the auto.py applies the default encoding of utf-8 to 5 types of files (txt, pdf, eml, html, xml)

`txt`: already implemented

`pdf`: The encoding kwarg is passed to partition_pdf, but is not used. (https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/pdf.py#L267) so I think the encoding kwarg should be removed from partition_pdf. 

`eml`: The .eml files have encoding information in the metadata. The encoding information is typically specified in the Content-Type header of the email message. (https://github.com/Unstructured-IO/unstructured/blob/main/example-docs/email-with-image.eml#L13). 

I think the encoding information should be parsed from the metadata if the user has not specified an encoding.

`html`: two options for filename and file-like object, except url
	option1: parse encoding information from html file metadata (<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">)
	option2: implement same pattern as partition_text

	factor to consider: whether to accept html files don't have charset information in the metadata

`xml`: similar to hmtl
	option1: parse encoding information from html file metadata (<?xml version="1.0" encoding="UTF-8"?>)
	option2: implement same pattern as partition_text

	factor to consider: whether to accept xml files don't have encoding information in the metadata
	
I'm not sure if I should do all the work for the 5 types of files in the same PR or if I should create separate PRs.

**Review by cragwolfe — APPROVED** [review]
LGTM!

**Pull request merged**